src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -259,6 +273,18 @@ def __call__(
             else:
                 latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs).prev_sample
 
+            # call the callback, if provided
+            if callback is not None:
+                if (callback_frequency is None) or (callback_frequency is not None and i % callback_frequency == 0):
+                    # scale and decode the image latents with vae
+                    current_latents = 1 / 0.18215 * latents
+                    image = self.vae.decode(current_latents).sample
+
+                    image = (image / 2 + 0.5).clamp(0, 1)
+                    image = image.cpu().permute(0, 2, 3, 1).numpy()
```

Comment:
I think this would be a great thing to extract to a method, as the exact same operations are done just outside the loop for the final results. (`decode_latents` was what I ended up calling it when I did this for similar reasons.)

I know that diffusers has a one-class-per-pipeline philosophy, and here's where I'm hoping that doesn't mean it's also a one-_method_-per-pipeline rule.